### DISCUSSION TITLE
Namespace Access Modifier

### DISCUSSION THREAD

**Discussion by joshfisk** [dotnet/csharplang#4987] [Language Ideas] [answered]
I sometimes see programs split into separate assemblies to make parts inaccessible from each other, or instead things are hidden in an implementation namespace. I think it would be easier to organize things in a single assembly, if you could use the namespace as a smaller enforced boundary within the assembly. Something with this new access modifier would only be visible in the same, and nested namespaces, but otherwise be treated as internal. Maybe it could be called `private internal`.

**Comment by yaakov-h** [comment, marked as answer]
Duplicate of #1138 and #2497.
